pkg/instancetype/upgrade_test.go

```diff
@@ -0,0 +1,585 @@
+				func(vm *virtv1.VirtualMachine) (*appsv1.ControllerRevision, error) {
+					preferenceSpec := instancetypev1alpha1.VirtualMachinePreferenceSpec{
+						CPU: &instancetypev1alpha1.CPUPreferences{
+							PreferredCPUTopology: instancetypev1alpha1.PreferSockets,
+						},
+					}
+					specBytes, err := json.Marshal(&preferenceSpec)
+					Expect(err).ToNot(HaveOccurred())
+
+					specRevision := instancetypev1alpha1.VirtualMachinePreferenceSpecRevision{
+						APIVersion: instancetypev1alpha1.SchemeGroupVersion.String(),
+						Spec:       specBytes,
+					}
+					specRevisionBytes, err := json.Marshal(specRevision)
+					Expect(err).ToNot(HaveOccurred())
+
+					return &appsv1.ControllerRevision{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:            "VirtualMachinePreferenceSpecRevision",
+							Namespace:       vm.Namespace,
+							OwnerReferences: []metav1.OwnerReference{*metav1.NewControllerRef(vm, virtv1.VirtualMachineGroupVersionKind)},
+						},
+						Data: runtime.RawExtension{
+							Raw: specRevisionBytes,
+						},
+					}, nil
+				},
+				updatePreferenceMatcher,
+			),
+			Entry("VirtualMachineClusterPreference v1alpha2",
+				func(vm *virtv1.VirtualMachine) (*appsv1.ControllerRevision, error) {
+					return CreateControllerRevision(vm,
+						&instancetypev1alpha2.VirtualMachineClusterPreference{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "clusterpreference",
+							},
+							Spec: instancetypev1alpha2.VirtualMachinePreferenceSpec{
+								CPU: &instancetypev1alpha2.CPUPreferences{
+									PreferredCPUTopology: instancetypev1alpha2.PreferSockets,
+								},
+							},
+						},
+					)
+				},
+				updatePreferenceMatcher,
+			),
+			Entry("VirtualMachineClusterPreference v1alpha1",
+				func(vm *virtv1.VirtualMachine) (*appsv1.ControllerRevision, error) {
+					return CreateControllerRevision(vm,
+						&instancetypev1alpha1.VirtualMachineClusterPreference{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: "clusterpreference",
+							},
+							Spec: instancetypev1alpha1.VirtualMachinePreferenceSpec{
+								CPU: &instancetypev1alpha1.CPUPreferences{
+									PreferredCPUTopology: instancetypev1alpha1.PreferSockets,
+								},
+							},
+						},
+					)
+				},
+				updatePreferenceMatcher,
+			),
+		)
+
+		DescribeTable("upgrade ControllerRevision without object version labels containing", func(createOriginalCR func(*virtv1.VirtualMachine) (*appsv1.ControllerRevision, error), updateMatcher func(*virtv1.VirtualMachine, string)) {
+			originalCR, err := createOriginalCR(vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			updateMatcher(vm, originalCR.Name)
+
+			if originalCR.Data.Object != nil {
+				originalCR.Data.Raw, err = json.Marshal(originalCR.Data.Object)
+				Expect(err).ToNot(HaveOccurred())
+			}
+			Expect(k8sClient.Tracker().Add(originalCR)).To(Succeed())
+
+			expectControllerRevisionCreation()
+			expectControllerRevisionDeletion(originalCR.Name)
```

Comment:
You removed the deletion of CRs with the same name, this will not pass?

Reply:
The removal of the original CR still happens when we switch things out to a new CR.